network_provisioning/include/network_provisioning/network_config.h

```diff
@@ -0,0 +1,207 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _NETWORK_PROV_CONFIG_H_
+#define _NETWORK_PROV_CONFIG_H_
+
+#include "esp_netif_ip_addr.h"
+#include "esp_err.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if CONFIG_ESP_WIFI_ENABLED
+/**
+ * @brief   WiFi STA status for conveying back to the provisioning master
+ */
+typedef enum {
+    NETWORK_PROV_WIFI_STA_CONNECTING,
+    NETWORK_PROV_WIFI_STA_CONNECTED,
+    NETWORK_PROV_WIFI_STA_DISCONNECTED
+} network_prov_wifi_sta_state_t;
+
+/**
+ * @brief   WiFi STA connection fail reason
+ */
+typedef enum {
+    NETWORK_PROV_WIFI_STA_AUTH_ERROR,
+    NETWORK_PROV_WIFI_STA_AP_NOT_FOUND
+} network_prov_wifi_sta_fail_reason_t;
+
+/**
+ * @brief   WiFi STA connected status information
+ */
+typedef struct {
+    /**
+     * IP Address received by station
+     */
+    char    ip_addr[IP4ADDR_STRLEN_MAX];
+
+    char    bssid[6];   /*!< BSSID of the AP to which connection was estalished */
+    char    ssid[33];   /*!< SSID of the to which connection was estalished */
+    uint8_t channel;    /*!< Channel of the AP */
+    uint8_t auth_mode;  /*!< Authorization mode of the AP */
+} network_prov_wifi_sta_conn_info_t;
+
+/**
+ * @brief   WiFi status data to be sent in response to `get_status` request from master
+ */
+typedef struct {
+    network_prov_wifi_sta_state_t wifi_state;    /*!< WiFi state of the station */
+    union {
+        /**
+         * Reason for disconnection (valid only when `wifi_state` is `WIFI_STATION_DISCONNECTED`)
+         */
+        network_prov_wifi_sta_fail_reason_t fail_reason;
+
+        /**
+         * Connection information (valid only when `wifi_state` is `WIFI_STATION_CONNECTED`)
+         */
+        network_prov_wifi_sta_conn_info_t   conn_info;
+    };
+} network_prov_config_get_wifi_data_t;
+
+/**
+ * @brief   WiFi config data received by slave during `set_config` request from master
+ */
+typedef struct {
+    char    ssid[33];       /*!< SSID of the AP to which the slave is to be connected */
+    char    password[64];   /*!< Password of the AP */
+    char    bssid[6];       /*!< BSSID of the AP */
+    uint8_t channel;        /*!< Channel of the AP */
+} network_prov_config_set_wifi_data_t;
+#endif // CONFIG_ESP_WIFI_ENABLED
+
+#if CONFIG_OPENTHREAD_ENABLED
+/**
+ * @brief   Thread status for conveying back to the provisioning master
+ */
+typedef enum {
+    NETWORK_PROV_THREAD_ATTACHING,
+    NETWORK_PROV_THREAD_ATTACHED,
+    NETWORK_PROV_THREAD_DETACHED,
+} network_prov_thread_state_t;
+
+/**
+ * @brief   Thread attaching fail reason
+ */
+typedef enum {
+    NETWORK_PROV_THREAD_DATASET_INVALID,
+    NETWORK_PROV_THREAD_NETWORK_NOT_FOUND,
+} network_prov_thread_fail_reason_t;
+
+/**
+ * @brief   Thread attached status information
+ */
+typedef struct {
+    uint32_t pan_id; /*!< PAN ID */
+    uint16_t channel; /*!< Channel */
+    uint8_t network_key[16]; /*!< Network Key */
```

Comment:
should not include the `network_key` in the status info

Reply:
removed